rotkehlchen/api/rest.py

```diff
@@ -4413,3 +4417,63 @@ def reprocess_skipped_external_events(self) -> Response:
             result=_wrap_in_ok_result({'total': total, 'successfull': successfull}),
             status_code=HTTPStatus.OK,
         )
+
+    def export_history_events(
+            self,
+            filter_query: HistoryBaseEntryFilterQuery,
+            directory_path: Optional[Path],
+    ) -> Response:
+        dbevents = DBHistoryEvents(self.rotkehlchen.data.db)
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            history_events, _, _ = dbevents.get_history_events_and_limit_info(
+                cursor=cursor,
+                filter_query=filter_query,
+                has_premium=True,
+                entries_limit=None,
+            )
+
+        if len(history_events) == 0:
+            return api_response(
+                wrap_in_fail_result('No history processed in order to perform an export'),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        serialized_history_events = []
+        headers: dict[str, None] = {}
+        for event in history_events:
+            serialized_event = event.serialize_for_csv()
+            serialized_history_events.append(serialized_event)
+            # maintain insertion order without storing extra info
+            headers.update({key: None for key in serialized_event})
+
+        try:
+            with tempfile.TemporaryDirectory() as temp_dir:
+                # on download
```

Comment:
this is a bit ugly because you are creating a temp file even if you don't need it. Why not move 

```
                dict_to_csv_file(
                    file_path,
                    serialized_history_events,
                    headers.keys(),
                )

                if directory_path is None:
                    return send_file(
                        path_or_file=file_path,
                        mimetype='text/csv',
                        as_attachment=True,
                        download_name=FILENAME_HISTORY_EVENTS_CSV,
                    )
```

to an auxiliar function. and then call it in the both possible places but avoiding the creation of the temp folder always. I would define a function inside this method but I believe lefteris doesn't like them much. Using another method could be okey

Reply:
so I only create the temp_dir when necessary, but abstracting with the dict_to_csv_file was looking too tacky because of the error handling it'll still require